home/src/app/pages/goods/goodsCtrl.js

```diff
@@ -1,50 +1,94 @@
         var category_slug = $stateParams.category_slug;
-        var url = "/"+category_slug+"/"+good_slug;
+        var url = "/" + category_slug + "/" + good_slug;
 
-        vm.items = {}; 
+        vm.items = {};
         vm.getGood(url);
-        
+
         function getGood(url) {
             goodService.getGood(url).then(function (response) {
                 $scope.item = response.data;
+                console.log($scope.item);
                 $scope.images = response.data.images;
-                googleMapService.getAddress(response.data.location[0].lat, response.data.location[0].lon).then(function(response) {
-                    $scope.address = response.data.results[0].address_components[0].short_name;
-                }, function(error) {
+                googleMapService.getAddress(response.data.location[0].lat, response.data.location[0].lon).then(function (response) {
+                    $scope.addreses = response.data.results[0].address_components[0].short_name;
+                    $scope.addresses = response.data.results;
+
+                }, function (error) {
                     console.log(error);
                 })
-             }, function () {
+            }, function () {
                 console.log('Something wrong when get good');
-             });
+            });
         }
 
-        $scope.rating =  function() {
+        $scope.sendMessage = function () {
+            console.log($scope.user.id);
+            var fromUser = $scope.user.id;
+            var title = "Have a message from your post: " + $scope.item.title;
+            title += " from " + fromUser;
+            var location = "<br/><p><b>Location:</b></p>";
+            location += "<ul class='listWithTypeListOrder'>";
+
+            $('input:checkbox[name="location"]:checked').each(function () {
+                console.log($(this).val());
+                location += "<li>" + $(this).val() + "</li>"
+            })
+            location += "</ul>";
+            location += "<br/><p>From: <a href='#/profile/" + fromUser + "'><b>" + fromUser + "</b></a></p>";
+            $scope.message.title = title;
+            $scope.message.content += location;
+            $scope.message.content += "<p><a href='#/goods" + url + "'>Link to goods</a></p>";
+            $scope.message.forUser = $scope.item.seller.id;
+            InboxService.SendMailPost($scope.message).then(function (response) {
+                toastr.success("Your message has been sent. Thank you to you message.");
+                $('#contactSeller').modal('hide');
+                $scope.message = {};
+            }, function (error) {
+                toastr.error(error);
+            })
+
+
+        }
+
+        $scope.rating = function () {
             console.log("rating");
             console.log($scope.item.seller.avgRating);
             $scope.ratingClick = 1;
-            $timeout(function() {
+            $timeout(function () {
```

Comment:
Refresh avg rating for user. That is good solution but I don't know how to load a gain the avg rating of user.